lib/mongo/auth/scram/conversation.rb

```diff
@@ -247,6 +255,8 @@ def client_final
         # @since 2.0.0
         def client_key
           @client_key ||= hmac(salted_password, CLIENT_KEY)
+          user.client_key ||= @client_key
+          @client_key
```

Comment:
I don't think you need to return @client_key explicitly. It will either return the ``user.client_key`` value, if there is one, or the ``@client_key`` value after being assigned to ``user.client_key``.